spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java

```diff
@@ -466,11 +466,11 @@ else if (tolerateLowerPartitionsOnBroker) {
 				// In some cases, the above partition query may not throw an UnknownTopic..Exception for various reasons.
 				// For that, we are forcing another query to ensure that the topic is present on the server.
 				if (CollectionUtils.isEmpty(partitions)) {
-					final AdminClient adminClient = AdminClient
-							.create(this.adminClientProperties);
-					final DescribeTopicsResult describeTopicsResult = adminClient
+					try (AdminClient adminClient = AdminClient
+							.create(this.adminClientProperties)){
```

Comment:
Thanks for the contribution.

Please run `./mvnw clean package` before submitting a PR; you'll see there are style violations (missing space before `{`; trailing spaces on line 473).

Reply:
I will fix these when merging this time; thanks again.